unit_tests/test_sct_events_grafana.py

```diff
@@ -65,14 +71,16 @@ def test_grafana(self):
             grafana_aggregator.time_window = 1
 
             set_grafana_url("http://localhost", _registry=self.events_processes_registry)
-
             with unittest.mock.patch("requests.post") as mock:
                 for runs in range(1, 4):
                     with self.wait_for_n_events(grafana_annotator, count=10, timeout=1):
                         for _ in range(10):
                             self.events_main_device.publish_event(
                                 ClusterHealthValidatorEvent.NodeStatus(severity=Severity.NORMAL))
                     time.sleep(1)
+                self.assertEqual(mock.call_count, 0)
+
+                start_posting_grafana_annotations(_registry=self.events_processes_registry)
 
                 self.assertEqual(mock.call_count, runs * 5)
```

Comment:
Unittests are failing here,

since it takes time to process the queue, this should be done with retrying

Reply:
Fixed.

Reply:
sleep might make this test flaky, I rather wait until `mock.call_count == runs * 5` with a timeout.

Reply:
We do not do here real network calls. We do here only local small computations. And 100ms is eternity for these ones.
And, in general, unit tests must not have sleeps at all. I just made the minimum changes to make the test work and cover the functionality.

Reply:
our event system is using networking (local) even in those unittests. but I do agree we should be doing sleeping.
that's why I don't this PR to be adding one more.

Reply:
Ok, added waiter.